File: app/src/main/java/org/cuieney/videolife/common/base/BaseBackFragment.java
package org.cuieney.videolife.common.base;

import android.support.v7.widget.Toolbar;

import org.cuieney.videolife.R;
import org.cuieney.videolife.common.component.EventUtil;

import me.yokeyword.fragmentation.SupportFragment;


/**
 * Created by cuieney on 16/2/7.
 */
public abstract class BaseBackFragment extends SupportFragment {
    protected static final String ARG_ITEM = "arg_item";

    protected void initToolbarNav(Toolbar toolbar) {
        toolbar.setNavigationIcon(R.mipmap.ic_arrow_back_white_24dp);
        toolbar.setNavigationOnClickListener(v -> {
            _mActivity.onBackPressed();
        });
        toolbar.inflateMenu(R.menu.video_menu);
    }

    @Override
    public boolean onBackPressedSupport() {
        EventUtil.sendEvent(false + "");
        return super.onBackPressedSupport();

    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/base/BaseBackFragment.kt
package org.cuieney.videolife.kotlin.base

import android.support.v7.widget.Toolbar
import me.yokeyword.fragmentation.SupportFragment
import org.cuieney.videolife.R
import org.cuieney.videolife.common.component.EventUtil

/**
 * Created by cuieney on 2017/5/22.
 */
abstract class BaseBackFragment : SupportFragment() {

    protected fun initToolbarNav(toolbar: Toolbar) {
        toolbar.setNavigationIcon(R.mipmap.ic_arrow_back_white_24dp)
        toolbar.setNavigationOnClickListener { v -> _mActivity.onBackPressed() }
        toolbar.inflateMenu(R.menu.video_menu)

    }

    override fun onBackPressedSupport(): Boolean {
        EventUtil.sendEvent(false.toString() + "")
        return super.onBackPressedSupport()

    }

    companion object {
        val ARG_ITEM = "arg_item"
    }
}